PHPCI/Application.php

```diff
@@ -86,31 +86,24 @@ public function handleRequest()
     {
         try {
             $this->response = parent::handleRequest();
-        } catch (HttpException $ex) {
-            $this->config->set('page_title', 'Error');
 
-            $view = new View('exception');
-            $view->exception = $ex;
+            if ($this->response->hasLayout() && $this->controller->layout) {
+                $this->setLayoutVariables($this->controller->layout);
+
+                $this->controller->layout->content  = $this->response->getContent();
+                $this->response->setContent($this->controller->layout->render());
+            }
 
-            $this->response->setResponseCode($ex->getErrorCode());
-            $this->response->setContent($view->render());
         } catch (\Exception $ex) {
             $this->config->set('page_title', 'Error');
 
             $view = new View('exception');
             $view->exception = $ex;
 
-            $this->response->setResponseCode(500);
+            $this->response->setResponseCode($ex instanceof HttpException ? $ex->getErrorCode() : 500);
```

Comment:
This is the only differences between the two catch blocks. If HttpException used/overrode Exception::getCode instead of defining its own getErrorCode, this checks wouldn't even be necessary.